src/storage-client/src/healthcheck.rs

```diff
@@ -36,8 +37,19 @@ pub fn pack_status_row(
     let collection_id = Datum::String(&collection_id);
     let status = Datum::String(status_name);
     let error = error.into();
-    let metadata = Datum::Null;
-    Row::pack_slice(&[timestamp, collection_id, status, error, metadata])
+
+    let mut row = Row::default();
+    let mut packer = row.packer();
+    packer.extend([timestamp, collection_id, status, error]);
+
+    match hint {
+        Some(hint) => {
+            let metadata = vec![("hint", Datum::String(hint))];
```

Comment:
in a followup pr, can you add docs explaining that `"hint"` might have extra info in the `details` column? docs can be found by searching around for where this is https://materialize.com/docs/sql/system-catalog/mz_internal/#mz_source_status_history in `doc/user`

Reply:
Will do 👍.

Would it make sense for it to be part of this PR? Or does usually docs is updated as a separate PR?

Reply:
separate pr is much easier to review!

Reply:
PR: https://github.com/MaterializeInc/materialize/pull/18150